### PR TITLE
Update cloudsearch api version and google api client version in order…

### PR DESCRIPTION
**Pull request by sarthak-kaps**
… to support request context boost feature. Fix the following unit tests that began breaking after the version update :

1) ApiOperationsTest
2) RepositoryExceptionTest
3) UploaderTest

`mvn clean install` is working correctly in root directory. All integration tests are running as expected locally.

API Version changes done per package : 
* [google-cloudsearch-indexing-connector-sdk] 
   - google-api-client 1.25.0 -> 2.2.0
* [google-cloudsearch-connector-sdk] 
   - google-api-client 1.25.0 -> 2.2.0
   - google-api-services-cloudsearch v1-rev18-1.25.0 -> v1-rev20230214-2.0.0
* [google-cloudsearch-connector-it-common]
   - google-api-client 1.25.0 -> 1.35.2
 * [google-cloudsearch-indentity-connector-sdk] 
   - Remove exclusion of google-api-client from google-api-services-admin-directory
   - Remove exclusion of google-api-client from google-api-services-cloudidentity

### CODE REVIEW

**Review by emailofanuj — APPROVED** [review]

File: identity/pom.xml [hunk 1]
```diff
@@ -122,10 +122,6 @@
           <groupId>com.google.guava</groupId>
           <artifactId>guava-jdk5</artifactId>
         </exclusion>
-        <exclusion>
-          <groupId>com.google.api-client</groupId>
```
Comment (emailofanuj) [inline identity/pom.xml:126]: just for understanding, why was this change needed?

File: indexing/pom.xml [hunk 2]
```diff
@@ -134,5 +134,10 @@
       <artifactId>JUnitParams</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.api-client</groupId>
+      <artifactId>google-api-client</artifactId>
+      <version>2.2.0</version>
```
Comment (emailofanuj) [inline indexing/pom.xml:140]: how did we decide on the version? Is this the latest version?

File: indexing/src/test/java/com/google/enterprise/cloudsearch/sdk/indexing/template/ApiOperationsTest.java [hunk 3]
```diff
@@ -128,10 +128,9 @@ public void pushItems_helpers() {
         .build();
     assertThat(pushItems.getPushItemResources().size(), equalTo(3));
     assertThat(pushItems.getPushItemResources().stream()
-        .map(item -> item.getId()).collect(ImmutableList.toImmutableList()),
-        equalTo(ImmutableList.of("item1", "item2", "item3")));
-    assertThat(pushItems.toString(), equalTo("PushItems [items=[[itemId=item1, pushItem={}], "
-            + "[itemId=item2, pushItem={queue=test queue}], [itemId=item3, pushItem={}]]]"));
+            .map(item -> item.getId()).collect(ImmutableList.toImmutableList()),
+            equalTo(ImmutableList.of("item1", "item2", "item3")));
+    assertThat(pushItems.toString(), equalTo("PushItems [items=[[itemId=item1, pushItem=GenericData{classInfo=[contentHash, metadataHash, payload, queue, repositoryError, structuredDataHash, type], {}}], [itemId=item2, pushItem=GenericData{classInfo=[contentHash, metadataHash, payload, queue, repositoryError, structuredDataHash, type], {queue=test queue}}], [itemId=item3, pushItem=GenericData{classInfo=[contentHash, metadataHash, payload, queue, repositoryError, structuredDataHash, type], {}}]]]"));
```
Comment (emailofanuj) [inline indexing/src/test/java/com/google/enterprise/cloudsearch/sdk/indexing/template/ApiOperationsTest.java:133]: Are we changing the response for the PushItem due to some parser upgrade? Why was this needed? Which dependency change caused this?

File: it/pom.xml [hunk 4]
```diff
@@ -153,5 +153,15 @@
       <artifactId>JUnitParams</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.apis</groupId>
+      <artifactId>google-api-services-cloudsearch</artifactId>
+      <version>v1-rev20230214-2.0.0</version>
```
Comment (emailofanuj) [inline it/pom.xml:159]: how did we decide this version?

File: it/pom.xml [hunk 5]
```diff
@@ -153,5 +153,15 @@
...
+    </dependency>
+    <dependency>
+      <groupId>com.google.api-client</groupId>
+      <artifactId>google-api-client</artifactId>
+      <version>1.35.2</version>
```
Comment (emailofanuj) [inline it/pom.xml:164]: how did we decide this version?

**Review by emailofanuj — CHANGES_REQUESTED** [review]

File: identity/pom.xml (hunk 1 above)
Comment (sarthak-kaps) [reply identity/pom.xml:126]: Making this change was allowing both `mvn clean install` and the integration tests to work correctly. I do not have a concrete explanation for now. This is a bit hacky. I think removing the exclusion is forcing maven to auto pick the appropriate dependency.

File: indexing/pom.xml (hunk 2 above)
Comment (sarthak-kaps) [reply indexing/pom.xml:140]: Yes this is the latest version of google-api-client.

File: indexing/src/test/java/com/google/enterprise/cloudsearch/sdk/indexing/template/ApiOperationsTest.java (hunk 3 above)
Comment (sarthak-kaps) [reply indexing/src/test/java/com/google/enterprise/cloudsearch/sdk/indexing/template/ApiOperationsTest.java:133]: Due to update in api version (not sure which one exactly) there was a change in PushItems response string.

File: it/pom.xml (hunk 4 above)
Comment (sarthak-kaps) [reply it/pom.xml:159]: This is the latest version.

File: it/pom.xml (hunk 5 above)
Comment (sarthak-kaps) [reply it/pom.xml:164]: Good question, I tried to make this 2.2.0 but this was breaking the identity integration tests. I had to settle for this version only since fixing the issue could take more time.
